### ISSUE TITLE
words containing ligatures show up in LOG and UHY files without the ligature

### ISSUE THREAD

**Issue by Mo-Gul** [pgundlach/lua-check-hyphen#13]
In short: Words that contain a ligature like "Differenz" are shown as "dierenz" in the LOG and as "die-renz" in the UHY file at least under Windows. (Ulrike Fischer confirmed that behavior with TL and MiKTeX.)

What follows is Ulrikes answer via email, where she had a look at the code (24.05.2016). Maybe that helps tracking down the issue:

> ich habe mal etwas in den lua-Code geschaut. Ich weiß nicht soviel über die ganzen node-Feinheiten, aber mein Eindruck ist, dass luacheckhyphen.collect_discs fehlerhaft ist, und die ff verliert.
> 
> Da ist zwar ein Zweig, der so aussieht, als wollte er Ligaturen einsammeln, aber der wird nie erreicht.
> 
> M.E. muss man in der Dokumentation noch mal genau den Abschnitt 4.5 Applying ligatures and kerning lesen und genau prüfen, wie die Nodes eigentlich aufgebaut sind, wenn die Funktion angewendet wird.
> 
> Ich kann mir ehrlich gesagt auch nicht vorstellen, dass es in Linux funktioniert, wenn dort ein aktuelles luatex > .95 benutzt wird.

**Comment by pgundlach** [comment]
A test file would help me to track this down. 

````
\documentclass{article}
\usepackage{fontspec}
\usepackage{lua-check-hyphen}
\parindent 0pt
\begin{document}

\hsize 1pt fluffiest

\end{document}
````

Looks OK on texlive 2015 / Mac

**Comment by Mo-Gul** [comment]
Yes, but it seems that it doesn't work when the ligature itself is hyphenated like in "Differenz"

```
\documentclass{article}
\usepackage{fontspec}
\usepackage{lua-check-hyphen}
    \parindent 0pt
\begin{document}

\hsize 1pt Differenz

\end{document}
```
which is then showing up as "Di-erenz" in the UHY file.

Interestingly 
```
\documentclass{article}
\usepackage{fontspec}
\usepackage{lua-check-hyphen}
    \parindent 0pt
\begin{document}
    \hsize 1pt Differenz
\end{document}
```
doesn't show any word in the UHY file. Extending this with other words only the last word is missing in that case.

**Comment by pgundlach** [comment]
TeXLive 2015 

The first code gives just the word "Differenz" in the output, 'Dif-ferenz'  in the log file and nothing in the uhy file is empty (somewhat strange).

The bug in the second code is unfortunate and documented in the manual. Any idea how to circumvent this is appreciated.

**Comment by pgundlach** [comment]
Just updated TL2015. The logfile is now `Dif-ferenz`, the output (PDF) is `Dif-ferenz` and the uhy file is `di-erenz`

So the only problem I can see is the missing f-f in the `uhy` file.

**Comment by pgundlach** [comment]
The function `collect_discs` should be re-written to take the pre/post/replace compartment into account. The attached PDF files have samples of nodelists. (`dif-fer-ence`, `fluffi-est` and `hy-phen-ation`)


[difference-nl.pdf](https://github.com/pgundlach/lua-check-hyphen/files/283792/difference-nl.pdf)
[fluffiest-nl.pdf](https://github.com/pgundlach/lua-check-hyphen/files/283793/fluffiest-nl.pdf)
[hyphenation-nl.pdf](https://github.com/pgundlach/lua-check-hyphen/files/283795/hyphenation-nl.pdf)

Thanks a lot for the report, but I can't state a ETA for the bugfix yet.

**Comment by sieversMartin** [comment]
Just another example:
```latex
% !TeX TS-program = lualatex
\documentclass{book}
\usepackage{fontspec}
\usepackage[ngerman]{babel}
\usepackage{lua-check-hyphen}

\begin{document}
\parbox{2cm}{danebentreffen}
\end{document}
```

gives the correct hyphenation danebentref-fen in the PDF, but the log and uhy file say:
``danebentre-en``
